scripts/run_performance_job.py

```diff
@@ -421,56 +802,132 @@ def run_performance_job(args: RunPerformanceJobArgs):
-        work_item_timeout=timedelta(hours=4),
-        work_item_dir=performance_setup_data.work_item_directory,
-        correlation_payload_dir=performance_setup_data.payload_directory,
+        work_item_timeout=work_item_timeout,
+        work_item_dir=work_item_dir,
+        correlation_payload_dir=payload_dir,
         project_file=args.project_file,
-        build_config=performance_setup_data.build_config,
+        build_config=build_config,
         performance_repo_dir=args.performance_repo_dir,
-        system_access_token=args.system_access_token,
         helix_build=args.build_number,
-        dotnet_cli_package_type="",
-        dotnet_cli_version="",
-        enable_xunit_reporter=False,
-        helix_prereq_commands=[],
-        include_dotnet_cli=False,
-        wait_for_work_item_completion=True,
         partition_count=args.partition_count,
-        cli_arguments=cli_arguments,
-        runtime_flavor=args.runtime_flavor or ""
-    )
+        runtime_flavor=args.runtime_flavor or "",
+        hybrid_globalization=args.hybrid_globalization,
+        target_csproj=args.target_csproj,
+        work_item_command=work_item_command,
+        bdn_arguments=bdn_arguments,
+        download_files_from_helix=True,
+        targets_windows=args.os_group == "windows",
+        helix_results_destination_dir=helix_results_destination_dir,
+        python=python,
+        affinity=args.affinity)
+    
+    if args.send_to_helix:
+        perf_send_to_helix(perf_send_to_helix_args)
 
-    perf_send_to_helix(perf_send_to_helix_args)
+        results_glob = os.path.join(helix_results_destination_dir, '**', '*perf-lab-report.json')
+        all_results: list[Any] = []
+        for result_file in glob(results_glob, recursive=True):
+            with open(result_file, 'r', encoding="utf8") as report_file:
+                all_results.extend(json.load(report_file))
 
-    results_glob = os.path.join(performance_setup_data.payload_directory, "performance", "artifacts", "helix-results", '**', '*perf-lab-report.json')
-    all_results: list[Any] = []
-    for result_file in glob(results_glob, recursive=True):
-        with open(result_file, 'r', encoding="utf8") as report_file:
-            all_results.extend(json.load(report_file))
+        output_counters_for_crank(all_results)
+    else:
+        # expose environment variables to CI for sending to helix
+        perf_send_to_helix_args.set_environment_variables(save_to_pipeline=True)
 
-    output_counters_for_crank(all_results)
+        # _BuildConfig is used by CI during log publishing
+        set_environment_variable("_BuildConfig", build_config, save_to_pipeline=True) 
 
+        
 
-def main(argv: list[str]):
+def main(argv: List[str]):
     args: dict[str, Any] = {}
 
     i = 1
     while i < len(argv):
         key = argv[i]
         bool_args = {
             "--internal": "internal",
+            "--physical-promotion": "physical_promotion_run_type",
```

Comment:
Is there a reason we are doing our own arg parsing here instead of using the arg parsing library?

Reply:
I went with this approach as argparse is not compatible with python type hints/dataclasses, which I think is more valuable here. With this your python editor will be able to give you type errors if you are not using your args correctly.